geopmdpy/geopmdpy/session.py

```diff
@@ -162,21 +280,32 @@ def run(self, run_time, period, pid, print_header,
                                  CSV data.
 
             request_stream (typing.IO): Input from user describing the
-                                   requests to read.
+                                        requests to read.
 
             out_stream (typing.IO): Stream where output from will be
-                               printed.
+                                    printed.
 
         """
-        requests = ReadRequestQueue(request_stream)
+        if session_io is not None:
```

Comment:
If session_io is not None assert that request_stream, out_stream and report_stream are all None.